contracts/apps/disputable/IAgreement.sol

```diff
@@ -0,0 +1,101 @@
+import "../../lib/arbitration/IArbitrable.sol";
+
+
+contract IAgreement is IArbitrable, IACLOracle {
+
+    event Signed(address indexed signer, uint256 settingId);
+    event ActionSubmitted(uint256 indexed actionId);
+    event ActionClosed(uint256 indexed actionId);
+    event ActionChallenged(uint256 indexed actionId, uint256 indexed challengeId);
+    event ActionSettled(uint256 indexed actionId, uint256 indexed challengeId);
+    event ActionDisputed(uint256 indexed actionId, uint256 indexed challengeId);
+    event ActionAccepted(uint256 indexed actionId, uint256 indexed challengeId);
+    event ActionVoided(uint256 indexed actionId, uint256 indexed challengeId);
+    event ActionRejected(uint256 indexed actionId, uint256 indexed challengeId);
+    event DisputableAppRegistered(address indexed disputable);
+    event DisputableAppUnregistered(address indexed disputable);
+
+    enum ChallengeState {
+        Waiting,
+        Settled,
+        Disputed,
+        Rejected,
+        Accepted,
+        Voided
+    }
+
+    function sign() external;
+
+    function newAction(uint256 _disputableActionId, bytes _context, address _submitter) external returns (uint256);
+
+    function closeAction(uint256 _actionId) external;
+
+    function challengeAction(uint256 _actionId, uint256 _settlementOffer, bool _finishedSubmittingEvidence, bytes _context) external;
+
+    function settle(uint256 _actionId) external;
+
+    function disputeAction(uint256 _actionId, bool _finishedSubmittingEvidence) external;
+
+    function register(
+        address _disputable,
+        ERC20 _collateralToken,
+        uint256 _actionAmount,
+        uint256 _challengeAmount,
+        uint64 _challengeDuration
+    )
+        external;
+
+    function unregister(address _disputable) external;
+
+    function getSigner(address _signer) external view returns (uint256 lastSettingIdSigned, bool mustSign);
+
+    function getAction(uint256 _actionId) external view
+        returns (
+            address disputable,
+            uint256 disputableActionId,
+            uint256 collateralId,
+            uint256 settingId,
+            address submitter,
+            bool closed,
+            bytes context,
+            uint256 currentChallengeId
+        );
+
+    function getChallenge(uint256 _challengeId) external view
+        returns (
+            uint256 actionId,
+            address challenger,
+            uint64 endDate,
+            bytes context,
+            uint256 settlementOffer,
+            uint256 arbitratorFeeAmount,
+            ERC20 arbitratorFeeToken,
+            ChallengeState state,
+            bool submitterFinishedEvidence,
+            bool challengerFinishedEvidence,
+            uint256 disputeId,
+            uint256 ruling
+        );
+
+    function getCurrentSettingId() external view returns (uint256);
+
+    function getSetting(uint256 _settingId) external view returns (string title, bytes content, IArbitrator arbitrator);
```

Comment:
> Just a thought, but wondering if it would also be worth storing the effective start time of the setting in the storage?

For now this is actually being effective immediately, we don't have a way to schedule it for a feature, are you suggesting to have it? If not, can't we rely on the block timestamp? actually storing the current timestamp is actually the same

Reply:
So I think this is an interesting getter—normally we wouldn't include the `bytes` and `string` arguments in the getter and log them in events instead.

But since:

1. This is likely to be used a lot from other contexts that have not reduced all the events or want to do an off-chain query (especially for the title)
1. Changing these settings is unlikely to happen very often
1. We will most likely want to display this information in the client without a long(er) load time to fetch all the events

It makes sense to keep the information we otherwise wouldn't find any use for on-chain.

And because of this, I thought it might be useful to also know through the getter when the Agreement was in effect.

Reply:
Discussed offline: similarly to other conclusions, we prefer not having this information off-chain vs spending 20k gas to have it since it can be at least retrieved off-chain